src/mcp_analyzer/checkers/security.py

```
@@ -0,0 +1,238 @@
+                    affected_component=target,
+                )
+            )
+
+        return findings
+
+    @staticmethod
+    def _check_network_exposure(parsed_url: httpx.URL) -> List[SecurityFinding]:
+        findings: List[SecurityFinding] = []
+        hostname = parsed_url.host or ""
+
+        if hostname in {"localhost", "127.0.0.1", "0.0.0.0"}:
```

Ruff (0.13.1)

Possible binding to all interfaces

(S104)